client/js/widgets/widget.js

```diff
@@ -2222,8 +2222,8 @@ export class Widget extends StateManaged {
       if(widgetType != 'card' && widgetType != 'pile')
         continue;
 
-      // check if this widget is closer than 10px from another widget in the same parent
-      if(widget.get('parent') == thisParent && Math.abs(widget.get('x')-thisX) < 10 && Math.abs(widget.get('y')-thisY) < 10) {
+      // check if this widget is closer than the pileSnapRange from another widget in the same parent
+      if(widget.get('parent') == thisParent && Math.abs(widget.get('x')-thisX) < this.get('pileSnapRange') && Math.abs(widget.get('y')-thisY) < this.get('pileSnapRange')) {
```

Comment:
So I find it a bit awkward that if you forget to specify this on `onPileCreation` that setting it in cardDefaults only changes it for dragging cards but not piles.

https://test.virtualtabletop.io/PR-1562/card-snap-range

Whereas if we forced the setting to be in `onPileCreation` or on piles, then you would guarantee that setting it sets it for both the cards and piles of cards.

The other strange thing is what @ArnoldSmith86 mentioned that this will use the snap range of the card your dragging and not care about the other's snap range so you'll get different behavior depending on which card you're dragging, e.g. in the following table if you drag the queen it has a large snap distance for snapping onto the king, and small if you drag the king to snap onto the queen:

https://test.virtualtabletop.io/PR-1562/different-snap

Reply:
Everything about piles is awkward.

>Whereas if we forced the setting to be in onPileCreation or on piles, then you would guarantee that setting it sets it for both the cards and piles of cards.

I don't think that is true.  At least not in my testing.  Single cards need a value in cardDefaults or directly on each individual card.  Piles need their own setting.  I don't know of a way to have the variable on cards and the variable on piles talk to each other and also be capable of being separate things.  I'm open to ideas?

> The other strange thing is what @ArnoldSmith86 mentioned that this will use the snap range of the card your dragging and not care about the other's snap range so you'll get different behavior depending on which card you're dragging, e.g. in the following table if you drag the queen it has a large snap distance for snapping onto the king, and small if you drag the king to snap onto the queen:

What do you propose as a solution?  Personally, I think it is unlikely in most cases that different cards would have different values.  This makes the most sense if you make piles and cards in the same deck all have the same value.  Of course, there are always other use cases.  Do we automatically make it so the engine uses the card being dragged (current behavior)?  Or the card/pile on the floor?  Or something else?  Or do we create some other property that lets the user set whether it is the dragging card or the one on the floor?

Reply:
Perhaps easier to show than to explain:
https://github.com/ArnoldSmith86/virtualtabletop/pull/1568

Then you can set this on the deck:
```
  "cardDefaults": {
    "onPileCreation": {
      "pileSnapRange": 100
    }
  },
```

And the same value applies to individual cards and piles created by those cards.

Reply:
Yes.  That works perfectly.  That is what I wanted to do but couldn't make it happen.